src/include/duckdb/common/vector_operations/generic_executor.hpp

```diff
@@ -67,23 +81,32 @@ struct StructTypeState {
 template <class A_TYPE>
 struct StructTypeUnary {
 	A_TYPE a_val;
+	bool is_null;
 
 	using STRUCT_STATE = StructTypeState<1>;
 
-	static bool ConstructType(STRUCT_STATE &state, idx_t i, StructTypeUnary<A_TYPE> &result) {
+	bool ContainsNull() {
+		return is_null;
```

Comment:
`ContainsNull` should return whether or not this field, or **any** child field is null, i.e. this should return `is_null || a_val.is_null` - same for all of the other elements